### ISSUE TITLE
Add workflow to detect link rot?

### ISSUE THREAD

**Issue by IndrajeetPatil** [krlmlr/actions-sync#34]
I am incubating this in `{dm}`: https://github.com/cynkra/dm/pull/1671

As can be seen there, this workflow can find the URLs that need to be updated. 

Link rot is probably one of the most common source of a `NOTE` in CRAN's incoming checks, and it'd be good to insure against it in the CI/CD pipeline.

--- 

Workflow:

```yaml
on:
  push:
    branches: [main, master]
  pull_request:
    branches: [main, master]

name: check-link-rot

jobs:
  check-link-rot:
    runs-on: ubuntu-latest
    env:
      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
      R_KEEP_PKG_SOURCE: yes
    steps:
      - uses: actions/checkout@v3

      - uses: r-lib/actions/setup-pandoc@v2

      - uses: r-lib/actions/setup-r@v2
        with:
          r-version: "devel"
          http-user-agent: "release"
          use-public-rspm: true

      - uses: r-lib/actions/setup-r-dependencies@v2
        with:
          pak-version: devel
          extra-packages: |
            any::rcmdcheck
            any::urlchecker
      - name: Run URL checker
        run: |
          options(crayon.enabled = TRUE)
          rotten_links <- urlchecker::url_check(progress = FALSE)
          print(rotten_links)
          if (length(rotten_links$URL) > 0L) {
            stop("Some URLs are outdated and need to be updated.", call. = FALSE)
          }
        shell: Rscript {0}
```

WDYT?

**Comment by IndrajeetPatil** [comment]
Actually, an even better way to handle this would be to make a PR with fixed URLs (using urlchecker::url_update(), the same way we do for snapshots.

**Comment by krlmlr** [comment]
Closed with #37, I believe.
